common/components/AddressBookTableRow.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import translate, { translateRaw } from 'translations';
+import noop from 'lodash/noop';
+import { Input, Identicon } from 'components/ui';
+
+interface Props {
+  index: number;
+  address: string;
+  label: string;
+  temporaryLabel: string;
+  labelError?: string;
+  isEditing: boolean;
+  onChange(label: string): void;
+  onSave(): void;
+  onLabelInputBlur(): void;
+  onEditClick(): void;
+  onRemoveClick(): void;
+}
+
+interface State {
+  labelInputTouched: boolean;
+}
+
+class AddressBookTableRow extends React.Component<Props> {
+  public state: State = {
+    labelInputTouched: false
+  };
+
+  private labelInput: HTMLInputElement | null = null;
+
+  public componentWillReceiveProps(nextProps: Props) {
+    this.setState({ label: nextProps.label, mostRecentValidLabel: nextProps.label });
+  }
+
+  public render() {
+    const {
+      address,
+      temporaryLabel,
+      labelError,
+      isEditing,
+      onEditClick,
+      onRemoveClick
+    } = this.props;
+    const { labelInputTouched } = this.state;
+    const trOnClick = isEditing ? noop : onEditClick;
+    const labelInputClassName = labelInputTouched && labelError ? 'invalid' : '';
+    const hashName = `${address}-hash`;
+    const labelName = `${address}-label`;
+
+    return (
+      <React.Fragment>
+        <div className="AddressBookTable-row" onClick={trOnClick}>
+          <div className="AddressBookTable-row-input">
+            <div className="AddressBookTable-row-input-wrapper">
+              <label htmlFor={hashName} className="AddressBookTable-row-input-wrapper-label">
+                {translate('ADDRESS')}
+              </label>
+              <Input name={hashName} title={address} value={address} readOnly={true} />
+            </div>
+            <div className="AddressBookTable-row-identicon AddressBookTable-row-identicon-non-mobile">
+              <Identicon address={address} />
+            </div>
+            <div className="AddressBookTable-row-identicon AddressBookTable-row-identicon-mobile">
+              <Identicon address={address} size="3rem" />
+            </div>
+          </div>
+          <div className="AddressBookTable-row-input">
+            <div className="AddressBookTable-row-input-wrapper">
+              <label htmlFor={labelName} className="AddressBookTable-row-input-wrapper-label">
+                {translate('LABEL')}
+              </label>
+              <Input
+                name={labelName}
+                title={`${translateRaw('EDIT_LABEL_FOR')}${address}`}
```

Comment:
There should be a space between these two.

Reply:
:+1: